src/app/icd/server/ICDMonitoringTable.cpp

```diff
@@ -242,15 +241,21 @@ CHIP_ERROR ICDMonitoringTable::Get(uint16_t index, ICDMonitoringEntry & entry) c
 
 CHIP_ERROR ICDMonitoringTable::Find(NodeId id, ICDMonitoringEntry & entry)
 {
-    uint16_t index = 0;
-    while (index < this->Limit())
+    CHIP_ERROR err;
+    uint16_t index;
+    ICDMonitoringEntry tempEntry(mSymmetricKeystore);
+
+    for (index = 0; index < this->Limit(); index++)
     {
-        ReturnErrorOnFailure(this->Get(index++, entry));
-        if (id == entry.checkInNodeID)
+        SuccessOrExit(err = this->Get(index, tempEntry));
```

Comment:
Consider not using SuccessOrExit/ GoTo label.
You could use continue; here

Reply:
Sure, thanks.
I've decided to use `break` as we want to get the first ID that has no entry assigned

Reply:
I've tested the change on desk and everything behaves correctly, so I'm resolving the comment

Reply:
Yes you are right break, as you want to stop the loop. Sorry!